src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -3109,15 +3109,15 @@ MulticopterPositionControl::task_main()
 
 			/* make sure attitude setpoint output "disables" attitude control
 			 * TODO: we need a defined setpoint to do this properly especially when adjusting the mixer */
-			_att_sp.thrust = 0.0f;
+			_att_sp.thrust_z = 0.0f;
 			_att_sp.timestamp = hrt_absolute_time();
 
 			/* reset velocity limit */
 			_vel_max_xy = _params.vel_max_xy;
 		}
 
 		/* reset setpoints and integrators VTOL in FW mode */
-		if (_vehicle_status.is_vtol && !_vehicle_status.is_rotary_wing) {
+		if (_vehicle_status.is_vtol && (!_vehicle_status.is_rotary_wing || _vehicle_status.in_transition_mode)) {
```

Comment:
@dagar This was required to reset the yaw_sp after a transition, otherwise the vehicle would mostly do a 180 in yaw.
Need to be done properly, other vtol types need the controller to run during transition.

Reply:
Oh okay, let's get it back in.